### PR TITLE
[Ruby 2.5] Add spec for Process#last_status

### PR DESCRIPTION
**Pull request by Atul9**
#565

### CODE REVIEW

File: core/process/last_status_spec.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ruby_version_is '2.5' do
+  describe 'Process#last_status' do
+    it 'returns the status of the last executed child process in the current thread' do
+      pid = Process.spawn("ruby", "-e", "exit 13")
+      Process.wait pid
+      Process.last_status.pid.should == pid
+    end
+
+    it 'returns nil if no child process has been ever executed in the current thread' do
+      Process.last_status.should == nil
```
Comment (eregon) [inline core/process/last_status_spec.rb]: This should be executed in a new thread so it's guaranteed there was no subprocess before.

File: core/process/last_status_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+
+ruby_version_is '2.5' do
+  describe 'Process#last_status' do
+    it 'returns the status of the last executed child process in the current thread' do
+      pid = Process.spawn("ruby", "-e", "exit 13")
```
Comment (eregon) [inline core/process/last_status_spec.rb]: This should use the current ruby (with `[*ruby_exe, '-e', 'exit 13']`) as there might not be a ruby in $PATH.
Alternatively, just `Process.spawn("exit 0")` should be portable and simpler (from #454).

File: core/process/last_status_spec.rb (hunk 1 above)
Comment (Atul9) [reply core/process/last_status_spec.rb]: How about if we run it before the spec 1?. I have updated the PR to run this nil check spec first.

Comment (eregon) [reply core/process/last_status_spec.rb]: `Process.last_status` like `$?` is thread-local I believe.
So if a previous spec used a subprocess (very likely) then it should fail.

Comment (Atul9) [reply core/process/last_status_spec.rb]: I am new to using the Process class. Regarding -- `executed in a new thread` I tried --
``` 
th = Thread.new do
   Process.last_status.should == nil
end
```
which outputs --
```
Expected #<Process::Status: pid 30790 exit 0>
 to equal nil```
How should we execute it in a new thread ?

File: core/process/last_status_spec.rb (hunk 2 above)
Comment (Atul9) [reply core/process/last_status_spec.rb]: done. I have updated the PR to use `Process.spawn("exit 0")`

File: core/process/last_status_spec.rb (hunk 1 above)
Comment (eregon) [reply core/process/last_status_spec.rb]: You can do something like:
```ruby
it "returns nil if no child process has been ever executed in the current thread" do
  Thread.new {
    Process.last_status.should == nil
  }.join
end
```
So the problem is the Thread was not `join`-ed which means the expectation might not have executed before the spec ends.

Comment (Atul9) [reply core/process/last_status_spec.rb]: Yes, that works. I have updated the PR.

**Comment by eregon** [comment]
Thank you for the new spec!

File: core/process/last_status_spec.rb (hunk 1 above)
Comment (eregon) [reply core/process/last_status_spec.rb]: Maybe activating the leak checker by default (`CHECK_LEAKS=t`) would help to find what goes wrong in such a case. I'll think about it.

**Pull request merged**